pikaur/install_info_fetcher.py

```diff
@@ -159,6 +159,15 @@ def get_all_packages_info(self) -> None:  # pylint:disable=too-many-branches
         # and their upgrades if --sysupgrade was passed
         if not self.args.aur:
             debug("Gonna get repo pkgs install info...")
+            # if pikaur is added as an argument for --sync
+            # add it to self.not_found_repo_pkgs_names
+            if ("pikaur" in self.install_package_names):
+                debug("Gonna add pikaur to packages to fetch from AUR...")
+                self.not_found_repo_pkgs_names.append('pikaur')
+                # we have to drop pikaur from the packages to install, otherwise
+                # since chaotic-aur/pikaur-git provides pikaur, we would install
+                # both, which is a conflict.
+                self.install_package_names.remove('pikaur')
```

Comment:
this "fix" is too specific, instead of checking package name you would need to check more general for that specific situation package origin and provides field, mb some other stuff

Reply:
Good point. Looking at the issue a bit more: If you insert the following after the self.get_repo_pkgs_info() just after my fix (or line 162 in the main repo):
```debug(print(self.repo_packages_install_info))```
it prints an empty list, as shown here:
```
~/b/p/pikaur (master±) ▶︎︎ ./pikaur.py -S pikaur --verbose --pikaur-debug
:: debug: Pikaur operation found for sys.argv=['./pikaur.py', '-S', 'pikaur', '--verbose', '--pikaur-debug']: cli_install_packages
:: debug: install_info_fetcher: Gonna fetch install info for:
    install_package_names=['pikaur']
    not_found_repo_pkgs_names=[]
    pkgbuilds_packagelists={}
    manually_excluded_packages_names=[]

:: debug: install_info_fetcher: Gonna get repo pkgs install info...
Reading repository package databases...
Reading local package database...
=> pacman --color=always --sync pikaur --print-format %r/%n
[]
```

If, however I run the command ```pacman --color=always --sync pikaur --print-format %r/%n``` outside of pikaur, it finds the pikaur-git package provided chaotic-aur is enabled:
```
~/b/p/pikaur (master±) ▶︎︎ pacman --color=always --sync pikaur --print-format %r/%n
chaotic-aur/pikaur-git
```

I would imagine this would happen for any package in a third party repo that is in a devel package but has a non-devel package available in the AUR. The only reason I checked for pikaur only is that that is the only that I'm aware causes this to arise.

Reply:
it have to do with Provides fields handling by pacman in repos - if in repos there are no package with exact match it will use the first one which Provides field have that package

so to fix it correctly - you need to check for the case if result from aur is exact match while result from repo not exact match - and in that case to use AUR package instead of repo one